### DISCUSSION TITLE
How to plot normalized rgb by time?

### DISCUSSION THREAD

**Discussion by zxdawn** [aazuspan/wxee#76] [Q&A] [answered]
Is it possible to normalize the rgb array by time dimension? I see if there're some clouds, the nomalization will cause some dark imgs like this:

```
ds_rgb.isel(time=slice(0, 5)).to_array(name="rgb").wx.normalize(stretch=0.99).plot.imshow(col='time')
```

<img width="1348" alt="image" src="https://github.com/user-attachments/assets/98258d41-61fe-45b6-9b41-f027a00d2c75">

**Comment by zxdawn** [comment, marked as answer]
Well, it's easier than I thought:

```
def normalize(da, stretch=1):
    min_val = np.quantile(da, 1-stretch)
    max_val = np.quantile(da, stretch)
    da_norm = np.clip((da - min_val) / (max_val - min_val), 0, 1)

    return da_norm


rgb = xr.apply_ufunc(normalize,
                     ds_rgb.isel(time=slice(0, 5)).to_array(name="rgb"),
                     exclude_dims=set(('variable', 'y', 'x')),
                     input_core_dims=[['variable', 'y', 'x']],
                     output_core_dims=[['variable', 'y', 'x']],
                     vectorize=True,
                     )
```

I can create a PR for this.

**Reply by aazuspan** [reply]
Glad you got it figured out so quickly! 

There's actually an even easier solution. If you calculate the quantiles only over the `x`, `y`, and `variable` dimensions (not `time`), you get one value per time coordinate and xarray will automatically broadcast everything correctly:

```python
def normalize(da, stretch = 1.0):
    min_val = da.quantile(1 - stretch, dim=["x", "y", "variable"])
    max_val = da.quantile(stretch, dim=["x", "y", "variable"])

    return ((da - min_val) / (max_val - min_val)).clip(0, 1)
```

Do you think this should be the default behavior for the `rgb` method? That seems like a good idea to me. If you want to make a PR, that'd be great!

EDIT: I originally suggested calculating quantiles over just `x` and `y`, but including `variable` like you did with the ufunc definitely looks better since it doesn't change relationships between bands.

**Reply by zxdawn** [reply]
Oh, yes! Thanks for your tip which is cleaner and better than mine ;) I will create a PR for RGB.
